### PR TITLE
NetSim: Mark as read

### PR DESCRIPTION
**Pull request by islemaster**
Adds some convenience features for students managing a large number of incoming packets:
- Packets are "unread" when they arrive, darker in color, but can be marked "read" by the student.  There's also an unread count in the log header.
- Packets can be collapsed and expanded.  Packets are always expanded when they arrive.  Collapsed packets show only the packet message, using the highest encoding available.

This might be styled differently in the future, but Baker says he likes the way it is now.

Unread packets screencast (from before the 'collapse' feature was in)

unread

Collapsing packets screencast

collapse_packets

### CODE REVIEW

File: apps/src/netsim/NetSimLogPacket.html.ejs [hunk 1]
```diff
@@ -56,68 +99,77 @@
...
+        logRow(EncodingType.DECIMAL,
+            binaryToInt(toAddress),
+            binaryToInt(fromAddress),
+            i18n.xOfYPackets({
+              x: binaryToInt(packetIndex),
+              y: binaryToInt(packetCount)
+            }),
+            alignDecimal(binaryToDecimal(message, chunkSize)));
+
+        logRow(EncodingType.HEXADECIMAL,
```
Comment (davidsbailey) [inline apps/src/netsim/NetSimLogPacket.html.ejs]: confused -- how are these log row calls (for encodings which are not represented in the incoming packet) not showing up in your ani gif?

Comment (islemaster) [inline apps/src/netsim/NetSimLogPacket.html.ejs]: :( this is probably not great, but they're hidden post-render by this `hideRowsByEncoding` call: https://github.com/code-dot-org/code-dot-org/pull/2034/files#diff-387834f5d915971245909e5dec678103R253

That's probably something I should change around to not render unneeded rows in the first place; the packet re-renders when encodings change anyway.

**Comment by davidsbailey** [comment]
LGTM - up to you whether you want to change the row hiding / rendering logic now or later. I'm assuming any additional changes re "just having the packet marked as read if you click anywhere in the packet" will be minor, so let me know if you want another look after.

**Comment by islemaster** [comment]
Removed the "mark as read" checkbox button, you click anywhere on the packet to mark it as read instead.  This is only a partial solution; tracking further visual cleanup as Pivotal item [#92387648](https://www.pivotaltracker.com/story/show/92387648).

**Comment by islemaster** [comment]
Tracking the log packet rendering cleanup as pivotal [#92387896](https://www.pivotaltracker.com/story/show/92387896)

**Pull request merged**